gb-ppu/examples/objects_viewer.rs

```diff
@@ -71,14 +84,21 @@ pub fn main() {
             egui::menu::bar(ui, |ui| {
                 ui.set_height(render::MENU_BAR_SIZE);
                 egui::menu::menu(ui, "dump", |ui| {
-                    for (title, vram_dump, oam_dump) in dumps {
+                    for (title, vram, oam, io_reg) in dumps {
                         if ui.button(title).clicked() {
-                            ppu.overwrite_vram(vram_dump);
-                            ppu.overwrite_oam(oam_dump);
-                            match title {
-                                "zelda" => ppu.control_mut().set_obj_size(true),
-                                _ => ppu.control_mut().set_obj_size(false),
-                            }
+                            ppu.overwrite_vram(vram);
+                            ppu.overwrite_oam(oam);
+                            *ppu.bg_palette_mut() = io_reg[0x47].into();
+                            *ppu.obj_palette_0_mut() = io_reg[0x48].into();
+                            *ppu.obj_palette_1_mut() = io_reg[0x49].into();
+                            ppu.control_mut()
+                                .set_win_tilemap_area((io_reg[0x40] & 0b0100_0000) != 0);
+                            ppu.control_mut()
+                                .set_bg_tilemap_area((io_reg[0x40] & 0b0000_1000) != 0);
+                            ppu.control_mut()
+                                .set_bg_win_tiledata_area((io_reg[0x40] & 0b0001_0000) != 0);
+                            ppu.control_mut()
+                                .set_obj_size((io_reg[0x40] & 0b0000_0100) != 0);
```

Comment:
Could you refactor this code into an utility fonction ?

Reply:
I've put this code in overwrite_memory, above the examples code.